src/post/post.service.ts

```diff
@@ -89,11 +89,25 @@ export class PostService {
                     : [attachment].filter((a) => a !== undefined);
                 for (const a of attachmentList) {
                     const attachmentJson = await a.toJsonLd();
+                    let url: URL;
+                    let mediaType: string;
+                    if (Array.isArray(attachmentJson.url)) {
+                        if (attachmentJson.url.length === 0) {
+                            // no usable URL – skip this attachment
+                            continue;
+                        }
+                        // attachments can have multiple urls, we need to handle this. For now we are using the first one.
+                        url = new URL(attachmentJson.url[0].href);
```

Comment:
Let's use the `parseURL` helper whenever we are parsing unknown data - this will catch the exception and return `null`. Then we can log an error or warning - but we don't need to crash the whole request